NumRush.py

```diff
@@ -38,165 +41,156 @@ def randomOperation(maxi):
+        display(strin, ResX // 2, 20)
     elif (pick == 3):
         strin = _("Q: %d / %d =") % (x, y)
-        texSurfaceObj = fontObj.render(strin, True, WHITE)
-        texRectObj = texSurfaceObj.get_rect()
-        texRectObj.center = (ResX / 2, 20)
-        DISPLAYSURF.blit(texSurfaceObj, texRectObj)
+        display(strin, ResX // 2, 20)
     elif (pick == 4):
         strin = _("Q: %d - %d =") % (x, y)
-        texSurfaceObj = fontObj.render(strin, True, WHITE)
-        texRectObj = texSurfaceObj.get_rect()
-        texRectObj.center = (ResX / 2, 20)
-        DISPLAYSURF.blit(texSurfaceObj, texRectObj)
+        display(strin, ResX // 2, 20)
     elif (pick == 5):
         strin = _("Q: %d - %d =") % (y, x)
-        texSurfaceObj = fontObj.render(strin, True, WHITE)
-        texRectObj = texSurfaceObj.get_rect()
-        texRectObj.center = (ResX / 2, 20)
-        DISPLAYSURF.blit(texSurfaceObj, texRectObj)
+        display(strin, ResX // 2, 20)
+    return strin
 
 
 def scoreBoard(score, hscore):
-    texSurfaceObj = fontObj.render(_("Score = %d") % (score), True, WHITE)
-    texRectObj = texSurfaceObj.get_rect()
-    texRectObj.center = (80, 20)
-    DISPLAYSURF.blit(texSurfaceObj, texRectObj)
+    display(_("Score = %d") % (score), 80, 20)
+    display(_("High Score = %d") % (hscore), ResX - 120, 20)
 
-    texSurfaceObj = fontObj.render(_("High Score = %d") % (hscore), True, WHITE)
-    texRectObj = texSurfaceObj.get_rect()
-    texRectObj.center = (ResX - 120, 20)
-    DISPLAYSURF.blit(texSurfaceObj, texRectObj)
 
+def newGameAnimation(mainGame):
 
-def newGameAnimation():
+    text = ['3', '2', '1', 'GO!']
 
-    DISPLAYSURF.fill(GREY)
-    texSurfaceObj = megaFont.render(("3"), True, BLACK)
-    texRectObj = texSurfaceObj.get_rect()
-    texRectObj.center = (ResX / 2, ResY / 2)
-    DISPLAYSURF.blit(texSurfaceObj, texRectObj)
-    pygame.display.update()
-    pygame.time.wait(1000)
-    DISPLAYSURF.fill(GREY)
-    texSurfaceObj = megaFont.render(("2"), True, BLACK)
-    texRectObj = texSurfaceObj.get_rect()
-    texRectObj.center = (ResX / 2, ResY / 2)
-    DISPLAYSURF.blit(texSurfaceObj, texRectObj)
-    pygame.display.update()
-    pygame.time.wait(1000)
-    DISPLAYSURF.fill(GREY)
-    texSurfaceObj = megaFont.render(("1"), True, BLACK)
-    texRectObj = texSurfaceObj.get_rect()
-    texRectObj.center = (ResX / 2, ResY / 2)
-    DISPLAYSURF.blit(texSurfaceObj, texRectObj)
-    pygame.display.update()
-    pygame.time.wait(1000)
-    DISPLAYSURF.fill(GREY)
-    texSurfaceObj = megaFont.render(_("GO!"), True, BLACK)
-    texRectObj = texSurfaceObj.get_rect()
-    texRectObj.center = (ResX / 2, ResY / 2)
-    DISPLAYSURF.blit(texSurfaceObj, texRectObj)
-    pygame.display.update()
-    pygame.time.wait(1500)
+    for i in text:
+
+        if mainGame.running == False:
+            return
+
+        DISPLAYSURF.fill(GREY)
+        texSurfaceObj = megaFont.render((i), True, BLACK)
```

Comment:
Probably the `(i)` parenthesis might be redundant here

Reply:
Actually, the code before the change used gettext to translate GO! but not to translate the digits 3, 2 and 1.  So this parenthesis is needed with a _ prefix for the last item in the list.  :smiley_cat:

Reply:
Ah good catch!